src/auth/components/birthdateForm/BirthdateForm.tsx

```diff
@@ -0,0 +1,132 @@
+import React from 'react';
+import { Field, Form, Formik } from 'formik';
+import { TextInput } from 'hds-react';
+import * as yup from 'yup';
+import { useTranslation } from 'react-i18next';
+import { differenceInYears } from 'date-fns';
+
+import styles from './BirthdateForm.module.css';
+import Button from '../../../common/button/Button';
+
+const schema = yup.object().shape({
+  birthDay: yup
+    .number()
+    .moreThan(-1)
+    .lessThan(32)
+    .required(),
+  birthMonth: yup
+    .number()
+    .moreThan(-1)
+    .lessThan(13)
+    .required(),
+  birthYear: yup
+    .number()
+    .moreThan(1000)
+    .required(),
+});
+
+type Props = {
+  checkBirthdate: (birthDate: string) => void;
+};
+
+type FormValues = {
+  birthDay: number | string;
+  birthMonth: number | string;
+  birthYear: number | string;
+};
+
+function BirthdateForm(props: Props) {
+  const { t } = useTranslation();
+
+  const isAgeValid = (values: FormValues) => {
+    if (
+      !values.birthDay ||
+      !values.birthMonth ||
+      !values.birthYear ||
+      values.birthYear < 1900
+    ) {
+      return false;
+    }
+    const age = differenceInYears(
+      new Date(),
+      new Date(
+        Number(values.birthYear),
+        Number(values.birthMonth),
+        Number(values.birthDay)
```

Comment:
Here `birthDay` can be `0` (based on `yup` validation), but the `Date` constructor expects the first date to be `1`. Only months has an index number.

Reply:
This is now removed.